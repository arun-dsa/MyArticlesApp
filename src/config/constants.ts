export const NAVIGATION_LINKS: { [key: string]: string } = {
  ROOT: "/",
  ARTICLES: "/articles",
  ARTICLE_DETAILS: "/articles/:id",
};
