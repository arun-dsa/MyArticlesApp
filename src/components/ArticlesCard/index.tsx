import { CardBodyText, CardTitleText } from "@components/StyledComponents";
import { ArticlesDetailsResponse } from "@interfaces/index";
import { Button, Card, CardActions, CardContent, Divider } from "@mui/material";
import { useNavigate } from "react-router-dom";
import ListAltIcon from "@mui/icons-material/ListAlt";
import ShareIcon from "@mui/icons-material/Share";
import { NAVIGATION_LINKS } from "@config/constants";

const ArticlesCard = ({ article }: { article: ArticlesDetailsResponse }) => {
  const navigate = useNavigate();

  const handleViewMore = () => {
    navigate(`${NAVIGATION_LINKS.ARTICLES}/${article.id}`);
  };

  return (
    <Card
      sx={{
        width: "100%",
        height: "100%",
        display: "flex",
        flexDirection: "column",
      }}
    >
      <CardContent sx={{ flex: 1 }}>
        <CardTitleText
          gutterBottom
          variant="h5"
          data-testid={`article-title-${article.id}`}
        >
          {article.title}
        </CardTitleText>
        <CardBodyText
          variant="body2"
          color="text.secondary"
          data-testid={`article-summary-${article.id}`}
        >
          {article.summary}
        </CardBodyText>
      </CardContent>
      <Divider />
      <CardActions sx={{ justifyContent: "flex-end" }}>
        <Button
          component="a"
          href={`mailto:change-it-to-email-to-share@gmail.com?subject=Read this article - ${article.title}&body=${article.summary}. Read the full article here ${window.location.href}articles/${article.id}`}
          size="small"
          variant="outlined"
          startIcon={<ShareIcon />}
          sx={{ textTransform: "none" }}
          data-testid={`share-article-${article.id}`}
        >
          Share
        </Button>
        <Button
          onClick={handleViewMore}
          size="small"
          variant="contained"
          startIcon={<ListAltIcon />}
          sx={{ textTransform: "none" }}
          data-testid={`article-view-full-${article.id}`}
        >
          View Full Article
        </Button>
      </CardActions>
    </Card>
  );
};

export default ArticlesCard;
