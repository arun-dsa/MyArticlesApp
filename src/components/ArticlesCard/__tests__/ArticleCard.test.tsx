import { ReactNode } from "react";
import { screen, render, fireEvent } from "@testing-library/react";
import { describe, expect, test } from "vitest";

import { routeWrapper } from "@utils/test_utils";
import { ARTICLES } from "@tests/mockData";
import ArticlesCard from "..";
import { NAVIGATION_LINKS } from "@config/constants";

describe("Article card test", () => {
  const article = ARTICLES[0];
  test("Test render article card", () => {
    const ArticleRoute: ReactNode = routeWrapper([
      {
        path: NAVIGATION_LINKS.ROOT,
        element: <ArticlesCard article={article} />,
      },
      {
        path: NAVIGATION_LINKS.ARTICLE_DETAILS,
        element: <>Article details page</>,
      },
    ]);

    render(ArticleRoute);

    expect(
      screen.getByTestId(`article-title-${article.id}`)
    ).toBeInTheDocument();
    expect(screen.getByTestId(`article-title-${article.id}`)).toHaveTextContent(
      article.title
    );

    expect(
      screen.getByTestId(`article-summary-${article.id}`)
    ).toBeInTheDocument();
    expect(
      screen.getByTestId(`article-summary-${article.id}`)
    ).toHaveTextContent(article.summary);

    expect(
      screen.getByTestId(`share-article-${article.id}`)
    ).toBeInTheDocument();

    expect(
      screen.getByTestId(`share-article-${article.id}`)
    ).toBeInTheDocument();
    expect(
      screen.getByTestId(`article-view-full-${article.id}`)
    ).toBeInTheDocument();

    fireEvent.click(screen.getByTestId(`article-view-full-${article.id}`));

    expect(screen.getByText("Article details page")).toBeInTheDocument();
  });

  test("Test snapshot of article card", () => {
    const ArticleRoute: ReactNode = routeWrapper([
      { path: "/", element: <ArticlesCard article={article} /> },
    ]);

    const container = render(ArticleRoute);
    expect(container).toMatchSnapshot();
  });
});
