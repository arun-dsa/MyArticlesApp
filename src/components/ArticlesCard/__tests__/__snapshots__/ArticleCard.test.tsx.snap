// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Article card test > Test snapshot of article card 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1rp1zv7-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-97bign-MuiCardContent-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1u4kd2-MuiTypography-root"
            data-testid="article-title-1"
          >
            Exploring the Wonders of the Great Barrier Reef
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1i87a3m-MuiTypography-root"
            data-testid="article-summary-1"
          >
            Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.
          </p>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1mgzbsd-MuiCardActions-root"
        >
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-nw1vud-MuiButtonBase-root-MuiButton-root"
            data-testid="share-article-1"
            href="mailto:change-it-to-email-to-share@gmail.com?subject=Read this article - Exploring the Wonders of the Great Barrier Reef&body=Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.. Read the full article here http://localhost:3000/articles/1"
            tabindex="0"
          >
            <span
              class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ShareIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92"
                />
              </svg>
            </span>
            Share
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-8fujfr-MuiButtonBase-root-MuiButton-root"
            data-testid="article-view-full-1"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ListAltIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5zm1.1-2H3.9c-.5 0-.9.4-.9.9v16.2c0 .4.4.9.9.9h16.2c.4 0 .9-.5.9-.9V3.9c0-.5-.5-.9-.9-.9M11 7h6v2h-6zm0 4h6v2h-6zm0 4h6v2h-6zM7 7h2v2H7zm0 4h2v2H7zm0 4h2v2H7z"
                />
              </svg>
            </span>
            View Full Article
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1rp1zv7-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-97bign-MuiCardContent-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1u4kd2-MuiTypography-root"
          data-testid="article-title-1"
        >
          Exploring the Wonders of the Great Barrier Reef
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1i87a3m-MuiTypography-root"
          data-testid="article-summary-1"
        >
          Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.
        </p>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-1mgzbsd-MuiCardActions-root"
      >
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-nw1vud-MuiButtonBase-root-MuiButton-root"
          data-testid="share-article-1"
          href="mailto:change-it-to-email-to-share@gmail.com?subject=Read this article - Exploring the Wonders of the Great Barrier Reef&body=Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.. Read the full article here http://localhost:3000/articles/1"
          tabindex="0"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ShareIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92"
              />
            </svg>
          </span>
          Share
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-8fujfr-MuiButtonBase-root-MuiButton-root"
          data-testid="article-view-full-1"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ListAltIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5zm1.1-2H3.9c-.5 0-.9.4-.9.9v16.2c0 .4.4.9.9.9h16.2c.4 0 .9-.5.9-.9V3.9c0-.5-.5-.9-.9-.9M11 7h6v2h-6zm0 4h6v2h-6zm0 4h6v2h-6zM7 7h2v2H7zm0 4h2v2H7zm0 4h2v2H7z"
              />
            </svg>
          </span>
          View Full Article
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Article details card test > Test render article details 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1rp1zv7-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-97bign-MuiCardContent-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1u4kd2-MuiTypography-root"
            data-testid="article-title-1"
          >
            Exploring the Wonders of the Great Barrier Reef
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1i87a3m-MuiTypography-root"
            data-testid="article-summary-1"
          >
            Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.
          </p>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1mgzbsd-MuiCardActions-root"
        >
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-nw1vud-MuiButtonBase-root-MuiButton-root"
            data-testid="share-article-1"
            href="mailto:change-it-to-email-to-share@gmail.com?subject=Read this article - Exploring the Wonders of the Great Barrier Reef&body=Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.. Read the full article here http://localhost:3000/articles/1"
            tabindex="0"
          >
            <span
              class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ShareIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92"
                />
              </svg>
            </span>
            Share
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-8fujfr-MuiButtonBase-root-MuiButton-root"
            data-testid="article-view-full-1"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ListAltIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5zm1.1-2H3.9c-.5 0-.9.4-.9.9v16.2c0 .4.4.9.9.9h16.2c.4 0 .9-.5.9-.9V3.9c0-.5-.5-.9-.9-.9M11 7h6v2h-6zm0 4h6v2h-6zm0 4h6v2h-6zM7 7h2v2H7zm0 4h2v2H7zm0 4h2v2H7z"
                />
              </svg>
            </span>
            View Full Article
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1rp1zv7-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-97bign-MuiCardContent-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1u4kd2-MuiTypography-root"
          data-testid="article-title-1"
        >
          Exploring the Wonders of the Great Barrier Reef
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1i87a3m-MuiTypography-root"
          data-testid="article-summary-1"
        >
          Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.
        </p>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-1mgzbsd-MuiCardActions-root"
      >
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-nw1vud-MuiButtonBase-root-MuiButton-root"
          data-testid="share-article-1"
          href="mailto:change-it-to-email-to-share@gmail.com?subject=Read this article - Exploring the Wonders of the Great Barrier Reef&body=Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.. Read the full article here http://localhost:3000/articles/1"
          tabindex="0"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ShareIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92"
              />
            </svg>
          </span>
          Share
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-8fujfr-MuiButtonBase-root-MuiButton-root"
          data-testid="article-view-full-1"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ListAltIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5zm1.1-2H3.9c-.5 0-.9.4-.9.9v16.2c0 .4.4.9.9.9h16.2c.4 0 .9-.5.9-.9V3.9c0-.5-.5-.9-.9-.9M11 7h6v2h-6zm0 4h6v2h-6zm0 4h6v2h-6zM7 7h2v2H7zm0 4h2v2H7zm0 4h2v2H7z"
              />
            </svg>
          </span>
          View Full Article
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
